skorch/scoring.py

```diff
@@ -0,0 +1,90 @@
+import numpy as np
+from skorch.net import NeuralNet
+from skorch.dataset import unpack_data
+
+
+def loss_scoring(net: NeuralNet, X, y=None):
+    """
+    loss_scoring(net, X, y=None)
+
+    Computes the loss of ``net`` on data (``X``, ``y``).
+
+    This function can be used to implement the ``score`` method for a
+    :class:`.NeuralNet` through sub-classing. This is useful, for example, when
+    combining skorch models with sklearn objects that rely on the model's
+    ``score`` method. Below is an example using GridSearchCV.
+
+    >>> class ScoredNet(skorch.NeuralNetClassifier):
```

Comment:
Thanks for adding the elaborate example. I think, however, that it could be simplified. 

1) You could just show how to subclass the net here.
2) You could go with the grid search example, but in that case you actually don't need to subclass the net. You can pass the `scoring=loss_scoring` to `GridSearchCV` and it should result in the same outcome.

Assume that you go with the second option: I would then add the subclassing example to the docs, probably to the FAQ ("How to implement a score method on the net that returns the loss?" or something along these lines).

Reply:
I wound up with a combination of 1. and 2. The docstring now shows the sub-classing. The FAQ now has an example of `loss_scoring(net, X, y)` as well as an example using `GridSearchCV`.